### DISCUSSION TITLE
Detecting Vanishing Point .. Return X and Y pixel location

### DISCUSSION THREAD

**Discussion by blakebullis** [ImageMagick/ImageMagick#4542] [Help]
I am looking to locate the "vanishing point(s)" of an image.  

Through googling, I have seen many recommend the Hough Lines approach. 

Example image using Hough Lines approach.  
http://www.fmwconcepts.com/imagemagick/houghlines/images/fence_lines_overlay50.png

I would like to take things a step further and identify the X and Y pixel location of where those red lines intersect.  I am not good with advanced ImageMagick scripts. 

Is there a known script to detect the vanishing point?

Is it possible for ImageMagick to return a JSON object (or another format) that could define the X,Y location in relation to the top-left corner of the image?

**Comment by snibgo** [comment]
If we have two non-parallel lines, we can readily find where they intersect.

If we have more than two lines that intersect at a single point, we can choose any pair to find that point.

If the lines _roughly_ intersect, we can calculate a point that minimizes the total error, as measured by (for example) the RMSE distances from that point to the lines.

When we have many lines, a subset of those lines may roughly intersect at one point, and another subset may roughly intersect at another point. And so on for other subsets. So a problem is to identify those subsets, then find each intersection.

This isn't a trivial problem. I am not aware of a scripted solution, but I expect someone somewhere has worked on it.

**Reply by blakebullis** [reply]
> If we have two non-parallel lines, we can readily find where they intersect.
> 
> If we have more than two lines that intersect at a single point, we can choose any pair to find that point.
> 
> If the lines _roughly_ intersect, we can calculate a point that minimizes the total error, as measured by (for example) the RMSE distances from that point to the lines.
> 
> When we have many lines, a subset of those lines may roughly intersect at one point, and another subset may roughly intersect at another point. And so on for other subsets. So a problem is to identify those subsets, then find each intersection.
> 
> This isn't a trivial problem. I am not aware of a scripted solution, but I expect someone somewhere has worked on it.

@snibgo  thank you for the reply.  I understand the concept of how the lines work to find the vanishing point.  I am looking for an automated/scripted solution though.  One that would return the X,Y coordinates, instead of an image with the overlayed lines.
